guest/rust/api/src/message.rs

```diff
@@ -0,0 +1,236 @@
+#[cfg(feature = "client")]
+/// Client-specific message functionality.
+pub mod client {
+    use crate::{
+        components::core::wasm::message::{data, source_module, source_network},
+        event,
+        global::{on, EntityId},
+        internal::{conversion::IntoBindgen, wit},
+        prelude::EventResult,
+    };
+
+    use super::Message;
+
+    #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+    /// Where a message came from.
+    pub enum Source {
+        /// This message came from the corresponding serverside module.
+        Network,
+        /// This message came from another serverside module.
+        Module(EntityId),
+    }
+
+    #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+    /// The target for a message.
+    pub enum Target {
+        /// An unreliable transmission to the server.
+        ///
+        /// Not guaranteed to be received, and must be below one kilobyte.
+        NetworkUnreliable,
+        /// A reliable transmission to the server (guaranteed to be received).
+        NetworkReliable,
+        /// A message to all other modules running on this client.
+        ModuleBroadcast,
```

Comment:
Think about other names to clarify that this is to other modules, not messaging yourself